resources/js/app.js

```diff
@@ -1,6 +1,37 @@
 (function(){
     var app = angular.module('coala', ['ngSanitize','btford.markdown', 'ngRoute']);
 
+    app.factory('Languages', function () {
+        var langs = [{
+            'name' : 'English',
+            'code' : 'en'
+        },
+        {
+            'name' : 'Bahasa',
+            'code' : 'id'
+        }];
+        var selected_language = langs[0];
+
+        return {
+        setData: function (val) {
+            angular.forEach(langs, function(v, k){
+                if (v.code == val){
+                    window.localStorage.setItem('lang', val);
+                    selected_language = v;
+                }
+            });
+        },
+        getData: function () {
+            if(selected_language.code)
+                return selected_language.code;
+            else return $location.search().lang || window.localStorage.getItem('lang');
+        },
+        getAllLanguages: function () {
+            return langs;
+            }
```

Comment:
this ``}`` should have same indentation level with ``getAllLanguages:...`` doesn't it?